src/main/kotlin/com/openlattice/data/ids/jobs/FixDuplicateIdAssignmentJob.kt

```diff
@@ -0,0 +1,301 @@
+                         * extra special bad and we do not know how to handle without manual intervention
+                         */
+                        logger.error("Woe is us, we will have to manually address.")
+                        recordCollision(id, entityKeys)
+                    }
+                }
+
+        state.idsProceessed += entityKeysAndIds.values.sumBy { it.size }
+        state.id = ids.max() ?: throw IllegalStateException("Entity key ids cannot be empty.")
+        hasWorkRemaining = (ids.size == BATCH_SIZE)
+    }
+
+    private fun assignNewIdsAndMoveData(originalId: UUID, entityKeys: List<EntityKey>) {
+        val newIds = idService.getNextIds(entityKeys.size).iterator()
+        hds.connection.use { connection ->
+            connection.autoCommit = false
+
+            val psSyncs = connection.prepareStatement(UPDATE_ID_ASSIGNMENT)
+            val psIds = connection.prepareStatement(MIGRATE_IDS)
+            val psData = connection.prepareStatement(MIGRATE_DATA)
+            val psEdges = connection.prepareStatement(MIGRATE_EDGES)
+            val psDst = connection.prepareStatement(UPDATE_E_DST)
+            val psEdge = connection.prepareStatement(UPDATE_E_EDGE)
+
+            entityKeys.forEach { entityKey ->
+                val newId = newIds.next()
+                updateIdAssignment(psSyncs, entityKey, newId)
+                migrate(psIds, entityKey.entitySetId, originalId)
+                migrate(psData, entityKey.entitySetId, originalId)
+                migrate(psEdges, entityKey.entitySetId, originalId)
+                update(psDst, newId, originalId)
+                update(psEdge, newId, originalId)
+            }
+            listOf(psSyncs,psIds, psData, psEdges, psDst, psEdge).forEach { it.executeBatch() }
+
+            connection.commit()
+        }
+    }
+
+    private fun migrate(
+            ps: PreparedStatement,
+            originalId: UUID,
+            entitySetId: UUID
+    ) {
+        bind(ps, entitySetId, originalId)
+        ps.addBatch()
+    }
+
+    private fun update(
+            ps: PreparedStatement,
+            newId: UUID,
+            originalId: UUID
+    ) {
+        bind(ps, newId, originalId)
+        ps.addBatch()
+    }
+
+    private fun bind(ps: PreparedStatement, entitySetId: UUID, entityKeyId: UUID) {
+        ps.setObject(1, entitySetId)
+        ps.setObject(1, entityKeyId)
+    }
+
+    private fun updateIdAssignment(ps: PreparedStatement, entityKey: EntityKey, newId: UUID) {
+        ps.setObject(1, newId)
+        ps.setObject(2, entityKey.entitySetId)
+        ps.setString(3, entityKey.entityId)
+    }
+
+    private fun recordCollision(id: UUID, entityKeys: List<EntityKey>) {
+        hds.connection.use { connection ->
+            connection.prepareStatement(INSERT_COLLISION).use {ps ->
+                ps.setObject(1, id )
+                entityKeys.forEach {
+                    ps.setObject(2, it)
```

Comment:
also looks like this update query should take entitySetId as param 2 and entityid as param 3